packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -111,7 +114,8 @@ let reducers: {
   },
   [ActionTypes.GoToOption](state, action) {
     if (state.disabled) return state
-    if (state.comboboxState === ComboboxStates.Closed) return state
+    if (!state.optionsPropsRef.current.static && state.comboboxState === ComboboxStates.Closed)
```

Comment:
I think that it might be okay for the combobox but not required. We can't do it for a Listbox / Menu because of focus trap / tab trap behaviour that is enabled based on that open / closed state. So maybe for that reason alone we shouldn't do it here either for consistency?